lms/djangoapps/courseware/views/views.py

```diff
@@ -1977,8 +1988,10 @@ def financial_assistance_request_v2(request):
         if request.user.username != username:
             return HttpResponseForbidden()
 
-        lms_user_id = request.user.id
         course_id = data['course']
+        if course_id and course_id not in request.META.get('HTTP_REFERER'):
```

Comment:
Can't a user edit from browser console request headers and change the id there?

Reply:
Right. What about the referrer policies? The user can tweak some settings in the browser(https://privacyinternational.org/guide-step/4149/change-http-referer-settings-chrome).

Reply:
Can't Django context variables be utilized for this purpose?

Reply:
I might have to look into what these context variables offer.

Reply:
Template variables.

Reply:
@Ali-D-Akbar Did you look into this? I am still not sold on referrer based approach as it is easily disruptable via third-party apps or browser config.

Reply:
@DawoudSheraz as discussed before, I'm limiting the select field to only have one value so the user can't change it by changing the html to remove the disable attribute.